### PR TITLE
c++: Implement subtree_index(...)

### PR DESCRIPTION
**Pull request by hunger**

### CODE REVIEW

File: internal/compiler/generator/cpp.rs [hunk 1]
```diff
@@ -966,13 +966,15 @@ fn generate_item_tree(
...
     target_struct.members.push((
         Access::Private,
         Declaration::Function(Function {
             name: "subtree_index".into(),
-            signature: "(slint::private_api::ComponentRef /* component */) -> uintptr_t".into(),
+            signature: "([[maybe_unused]] slint::private_api::ComponentRef component) -> uintptr_t"
+                .into(),
             is_static: true,
-            statements: Some(vec!["/* unimplemented! */".to_owned(), "return 0;".into()]),
+            statements: Some(vec![format!("return {};", std::usize::MAX)]),
```
Comment (ogoffart) [inline internal/compiler/generator/cpp.rs:977]: How about?

```suggestion
            statements: Some(vec!["return numeric_limits<uintptr_t>::max();".into()]),
```

Maybe better than a huge 64bit constant

Comment (hunger) [reply internal/compiler/generator/cpp.rs:977]: Nah! That would be way too readable! I'll change it.

File: internal/compiler/generator/cpp.rs [hunk 2]
```diff
@@ -1584,6 +1586,25 @@ fn generate_repeated_component(
         ));
     }
 
+    if let Some(index_prop) = repeated.index_prop {
+        // Override default subtree_index function implementation
+        let subtree_index_func = repeater_struct.members.iter_mut().find(|(_, d)| match d {
```
[the remark this answers is not in the document]
Comment (hunger) [reply internal/compiler/generator/cpp.rs:1591]: How so? I need that one specific item to override the implementation.

[the remark this answers is not in the document]
Comment (tronical) [reply internal/compiler/generator/cpp.rs:1591]: Ah right, sorry, I misread - disregard my comment :)

**Pull request merged**